x/encryption/encryption.go

```diff
@@ -0,0 +1,8 @@
+package encryption
+
+import "context"
+
+type Encryptor interface {
+	Encrypt(ctx context.Context, plainStr string) (encryptedStr *string, err error)
```

Comment:
just curious could we return string instead of *string
```suggestion
	Encrypt(ctx context.Context, plainStr string) (encryptedStr string, err error)
```